src/subapps/search/hooks/useSearchPagination.tsx

```diff
@@ -0,0 +1,182 @@
+import { NexusClient } from '@bbp/nexus-sdk';
+import * as React from 'react';
+import useMeasure from '../../../shared/hooks/useMeasure';
+import { debounce } from 'lodash';
+
+export type SearchPagination = {
+  numRowsFitOnPage: number;
+  currentPage: number;
+  totalNumberOfResults: number;
+  trueTotalNumberOfResults: number;
+  pageSize: number;
+  pageSizeOptions: string[];
+  pageSizeFixed: boolean;
+  isInitialized: boolean;
+};
+
+/* ElasticSearch's index.max_result_window setting (default to 10k)
+determines how far we can page into results */
+export const ESMaxResultWindowSize = 10000;
+export const defaultPageSizeOptions = [10, 20, 50, 100];
+
+function useSearchPagination(query: any, nexus: NexusClient) {
+  const defaulpagination: SearchPagination = {
+    numRowsFitOnPage: 0,
+    currentPage: 1,
+    totalNumberOfResults: 0,
+    trueTotalNumberOfResults: 0,
+    pageSize: 0,
+    pageSizeOptions: [] as string[],
+    pageSizeFixed: false,
+    isInitialized: false,
+  };
+  const [pagination, setPagination] = React.useState<SearchPagination>(
+    defaulpagination
+  );
+
+  const handlePaginationChange = (
+    page: number,
+    pageSize?: number | undefined
+  ) => {
+    setPagination(prevPagination => {
+      return {
+        ...prevPagination,
+        currentPage: page,
+        pageSizeFixed: prevPagination.pageSizeFixed,
+        pageSize: pageSize ? pageSize : prevPagination.pageSize,
+      };
+    });
+  };
+
+  const onShowSizeChange = React.useCallback(
+    (current, size) => {
+      if (defaultPageSizeOptions.includes(size)) {
+        localStorage.setItem('searchPageSize', size.toString());
+        setPagination(prevPagination => {
+          return {
+            ...prevPagination,
+            pageSizeFixed: true,
+          };
+        });
+      } else {
+        localStorage.removeItem('searchPageSize');
+        setPagination(prevPagination => {
+          return {
+            ...prevPagination,
+            pageSizeFixed: false,
+          };
+        });
+      }
+    },
+    [pagination]
+  );
+
+  const showTotal = React.useCallback(
+    (total: number, range: [number, number]) =>
+      pagination.trueTotalNumberOfResults <= ESMaxResultWindowSize ? (
+        <>
+          Showing {total.toLocaleString('en-US')}{' '}
```

Comment:
is it a way to convert a number to a string?

Reply:
Pagination stuff and height stuff is a copy/refactor from existing code.  But will address your comments any way.

Reply:
toLocaleString converts a number to string, but also makes sure that the string is dialect appropriate. for example, dates are written different in US vs Europe.